src/Build/Logging/BinaryLogger/BufferedBinaryReader.cs

```diff
@@ -0,0 +1,444 @@
+            FillBuffer(4);
+
+            var result = (int)(buffer[bufferOffset] | buffer[bufferOffset + 1] << 8 | buffer[bufferOffset + 2] << 16 | buffer[bufferOffset + 3] << 24);
+            bufferOffset += 4;
+            baseStreamPosition += 4;
+            return result;
+        }
+
+        // Reusable StringBuilder for ReadString().
+        private StringBuilder? cachedBuilder;
+
+        // Reusable char[] for ReadString().
+        private char[]? charBuffer;
+
+        /// <summary>
+        /// Reads a string with a prefixed of the length.
+        /// </summary>
+        /// <returns>A string.</returns>
+        public string ReadString()
+        {
+            int stringLength = Read7BitEncodedInt();
+            int stringOffsetPos = 0;
+            int readChunk = 0;
+
+            if (stringLength == 0)
+            {
+                return string.Empty;
+            }
+
+            if (stringLength < 0)
+            {
+                throw new Exception();
+            }
+
+            if (charBuffer == null)
+            {
+                charBuffer = new char[bufferCapacity + 1];
+            }
+
+            int charRead = 0;
+
+            if (bufferLength > 0)
+            {
+                // Read content in the buffer.
+                readChunk = stringLength < (bufferLength - bufferOffset) ? stringLength : bufferLength - bufferOffset;
+                charRead = encoding.GetChars(buffer, bufferOffset, readChunk, charBuffer, 0);
+                bufferOffset += readChunk;
+                baseStreamPosition += readChunk;
+                if (stringLength == readChunk)
+                {
+                    // if the string is fits in the buffer, then cast to string without using string builder.
+                    return new string(charBuffer, 0, charRead);
+                }
+                else
+                {
+                    cachedBuilder ??= new StringBuilder();
+                    cachedBuilder.Append(charBuffer, 0, charRead);
+                }
+            }
+
+            cachedBuilder ??= new StringBuilder();
+            stringOffsetPos += readChunk;
+
+            do
+            {
+                // Read up to bufferCapacity;
+                readChunk = Math.Min(stringLength - stringOffsetPos, bufferCapacity);
+                FillBuffer(readChunk);
+                charRead = encoding.GetChars(buffer, bufferOffset, readChunk, charBuffer, 0);
+                bufferOffset += readChunk;
+                baseStreamPosition += readChunk;
+                cachedBuilder.Append(charBuffer, 0, charRead);
+                stringOffsetPos += readChunk;
+            } while (stringOffsetPos < stringLength);
+
+            string result = cachedBuilder.ToString();
+            cachedBuilder.Clear();
+            return result;
+        }
+
+        /// <summary>
+        /// Reads an 8-byte signed integer.
+        /// </summary>
+        /// <returns></returns>
+        public long ReadInt64()
```

Comment:
Same here - where is it comming from? Is Big/Little Endian handled accordingly?

Reply:
Added note that it is from BCL.  I will need to check with the Endian, I believe the test will catch the issue on MacOS.

Reply:
Looks like endian doesn't matter as it is all little endian regardless of MacOS.  The binary writer should need be mindful to convert to little endian.